core/src/main/java/com/sequenceiq/cloudbreak/service/cost/InstanceTypeCollectorService.java

```diff
@@ -91,6 +108,38 @@ private Optional<InstanceGroupCostDto> getInstanceGroupCostDto(String region, Cl
         return Optional.empty();
     }
 
+    private Optional<InstanceGroupCO2Dto> getInstanceGroupCO2Dto(String region, CloudPlatform cloudPlatform,
+            InstanceGroupView instanceGroupView, Credential credential) {
+        if (pricingCacheMap.containsKey(cloudPlatform)) {
```

Comment:
we can expand this if statement to unclude instanceGroupView.getTemplate() != null, since I assume we do nothing useful if we can't get that info here.